app/models/course_template.rb

```diff
@@ -26,6 +27,9 @@ class CourseTemplate < ActiveRecord::Base
   scope :not_hidden, -> { where(hidden: false) }
   scope :available, -> { not_expired.not_hidden }
 
+  after_destroy :delete_courses_if_custom_courses_disabled
+  after_destroy :delete_cache
```

Comment:
I'm thinking that should course template deleting be allowed at all, if there are any courses made from that template? There's anyway hidden field plus expiredate for course templates already, to let old templates pass away.